llama-rs/src/lib.rs

```diff
@@ -185,43 +231,50 @@ pub enum LoadError {
     TensorWrongSize { tensor_name: String, path: PathBuf },
     #[error("invalid ftype {ftype} in {path:?}")]
     InvalidFtype { ftype: i32, path: PathBuf },
+    #[error("could not write model memory")]
+    FailedToWriteMemory(Box<dyn std::error::Error>),
+    #[error("could not read model memory")]
+    FailedToReadMemory(Box<dyn std::error::Error>),
```

Comment:
Could we avoid erasing the error type here by adding more cases to the variant and then using `Box<LlamaError>` instead?